inform7/Internal/Inter/BasicInformKit/Sections/StartUp.i6t

```diff
@@ -0,0 +1,184 @@
+These rules perform the essential tasks required for setting up the virtual
+machine. This includes checking that we are running in a compatible interpreter,
+and setting up various built in Glk objects (when running in a Glk interpreter).
+
+These rules here are the ones which get the basic machinery working
+to the point where it is safe to run arbitrary I7 source text. They necessarily
+do very low-level things, and it is not guaranteed that I7 phrases will behave
+to specification if executed before these early rules have finished. So it
+is hazardous to obstruct or alter them.
+
+Arrangements are a little different here on the Z-machine, because some
+data is retained in the case of a restart.
+
+(Many thanks are due to Eliuk Blau, who found several tricky timing errors
+here and elsewhere in the Glulx-specific code. Frankly, I feel like hanging
+a sign on the following routines which reads "Congratulations on bringing
+light to the Dark Room.")
+
+@h Before starting the virtual machine rules.
+The before starting the virtual machine rules run (in the case of Glulx/Glk)
+before any of the IO systems have been set up. This means that it is important
+that these rules do not print any text.
+
+=
+#Ifdef TARGET_GLULX;
+[ CHECK_FOR_GLK_R res;
+	@gestalt 4 2 res;
+	if (res == 0) {
+		quit;
+	}
+
+	unicode_gestalt_ok = false;
+	if (glk_gestalt(gestalt_Unicode, 0)) {
+		unicode_gestalt_ok = true;
+	}
+
+	! Set the VM's I/O system to be Glk.
+	@setiosys 2 0;
+	rfalse;
+];
+#Endif; ! TARGET_GLULX
+
+@h Capture startup text.
+It is a semi-common problem for authors to do something which results in text
+being printed before any Glk windows have been opened, which results in a fatal
+interpreter error. (Or even worse, an extension is the cause, which the author
+should not be expected to know the internal details of.) This is very frustrating,
+as the author often will not know what is the cause of the error as any error
+messages can't be shown until a Glk window is created; indeed it's often an error
+message that is trying to be printed, which the author will never see. To help
+authors, while we wait for the virtual machine to be set up (including a Glk
+window) we can instead send any text to a memory stream. Then once the VM is ready
+we can check if any text was sent to the memory stream, and then finally display
+it in the window. We only capture 256 bytes, which may not be all that is attempted
+to be printed, but hopefully it gives authors enough to identify the problem.
+
+=
+#Ifdef TARGET_GLULX;
+Constant PRE_STARTUP_TEXT_CAPTURE_MAX_LEN = 256;
+Array Pre_Startup_Text_Capture_Buffer -> PRE_STARTUP_TEXT_CAPTURE_MAX_LEN;
+Array Pre_Startup_Text_Capture_Results --> 2;
+Global Pre_Startup_Text_Capture_Stream;
+
+[ CAPTURE_STARTUP_TEXT;
+	Pre_Startup_Text_Capture_Stream = glk_stream_open_memory(Pre_Startup_Text_Capture_Buffer, PRE_STARTUP_TEXT_CAPTURE_MAX_LEN, filemode_Write, 0);
+	glk_stream_set_current(Pre_Startup_Text_Capture_Stream);
+];
```

Comment:
Just noticed that these functions don't `rfalse;`. I hate that I6 returns true if there's no explicit return statement...
I can't remember what returning true in a rulebook with default no decision outcome means, but it'd probably be safer to add the `rfalse;`s.